DEVELOPER.md

```diff
@@ -1,5 +1,48 @@
 # Developer Notes
 
+## Setup your development environment
+
+First fork https://github.com/m3db/m3 into your own workspace on Github.
+
+Then create your clone:
+
+```bash
+export working_dir=$GOPATH/src/github.com/m3db
+mkdir -p $workding_dir
+
+# Set this to your Github user
+export user={your github profile name}
```

Comment:
super nit: `export user="your_github_profile_name"` just so for those not super familiar with bash they know which quotes to use?

Reply:
Agreed, this might be better

Reply:
Fixed